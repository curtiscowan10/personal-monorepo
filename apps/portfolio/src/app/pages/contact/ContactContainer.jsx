import React, { useState } from 'react';
import {
  Email as EmailIcon,
  Phone as PhoneIcon,
  LocationOn as LocationIcon,
  LinkedIn as LinkedInIcon,
  GitHub as GitHubIcon,
  Twitter as TwitterIcon,
} from '@mui/icons-material';
import Contact from './Contact';

const ContactContainer = () => {
  const [formData, setFormData] = useState({
    name: '',
    email: '',
    subject: '',
    message: '',
  });
  const [showAlert, setShowAlert] = useState(false);

  const contactInfo = [
    {
      icon: <EmailIcon className="icon" />,
      title: 'Email',
      value: 'hello@portfolio.com',
      link: 'mailto:hello@portfolio.com',
    },
    {
      icon: <PhoneIcon className="icon" />,
      title: 'Phone',
      value: '+1 (555) 123-4567',
      link: 'tel:+15551234567',
    },
    {
      icon: <LocationIcon className="icon" />,
      title: 'Location',
      value: 'San Francisco, CA',
      link: null,
    },
  ];

  const socialLinks = [
    {
      icon: <LinkedInIcon />,
      name: 'LinkedIn',
      url: 'https://linkedin.com/in/portfolio',
      color: '#0077b5',
    },
    {
      icon: <GitHubIcon />,
      name: 'GitHub',
      url: 'https://github.com/portfolio',
      color: '#333',
    },
    {
      icon: <TwitterIcon />,
      name: 'Twitter',
      url: 'https://twitter.com/portfolio',
      color: '#1da1f2',
    },
  ];

  const handleInputChange = (e) => {
    const { name, value } = e.target;
    setFormData((prev) => ({
      ...prev,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    // Here you would typically send the form data to your backend
    console.log('Form submitted:', formData);
    setShowAlert(true);
    setFormData({
      name: '',
      email: '',
      subject: '',
      message: '',
    });
  };

  const handleCloseAlert = () => {
    setShowAlert(false);
  };

  return (
    <Contact
      formData={formData}
      showAlert={showAlert}
      contactInfo={contactInfo}
      socialLinks={socialLinks}
      onInputChange={handleInputChange}
      onSubmit={handleSubmit}
      onCloseAlert={handleCloseAlert}
    />
  );
};

export default ContactContainer;
