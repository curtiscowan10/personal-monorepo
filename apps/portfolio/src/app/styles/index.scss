// index.scss - Main styles entry point

// Import variables and mixins first
@use 'variables' as *;
@use 'mixins' as *;

// Global styles
* {
  box-sizing: border-box;
}

body {
  font-family: Inter, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto,
    sans-serif;
  margin: 0;
  padding: 0;
  line-height: 1.6;
  color: $color-primary;
  background-color: $color-white;
}

// Global utility classes following BEM-like naming
.u-text-center {
  @include text-center;
}

.u-text-truncate {
  @include text-truncate;
}

.u-text-gradient {
  @include text-gradient;
}

.u-fade-in {
  @include fade-in;
}

.u-slide-up {
  @include slide-up;
}

// Global component classes
.c-container {
  @include container;
}

.c-section {
  @include section-spacing;
}

.c-card {
  @include card;

  &--hover {
    @include card-hover;
  }

  &__content {
    @include card-content;
  }
}

.c-button {
  @include button-base;

  &--primary {
    @include button-primary;
  }

  &--secondary {
    @include button-secondary;
  }

  &--outline {
    @include button-outline;
  }
}

.c-form {
  &__input {
    @include form-input;
  }

  &__label {
    @include form-label;
  }
}

.c-grid {
  &--responsive {
    @include responsive-grid;
  }

  &--2-col {
    @include responsive-grid(1, 2, 2);
  }

  &--3-col {
    @include responsive-grid(1, 2, 3);
  }
}
