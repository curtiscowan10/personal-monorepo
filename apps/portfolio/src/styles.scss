/* Main styles entry point - imports the complete styling system */
@use 'app/styles/index';
