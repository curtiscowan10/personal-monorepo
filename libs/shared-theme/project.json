{
  "name": "shared-theme",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared-theme/src",
  "projectType": "library",
  "tags": []
}
