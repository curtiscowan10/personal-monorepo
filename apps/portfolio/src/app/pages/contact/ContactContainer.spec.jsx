import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { ThemeProvider } from '@mui/material/styles';
import { theme } from '@personal-monorepo/shared-theme';
import ContactContainer from './ContactContainer';

const renderWithProviders = (component) => {
  return render(
    <BrowserRouter>
      <ThemeProvider theme={theme}>{component}</ThemeProvider>
    </BrowserRouter>
  );
};

describe('ContactContainer', () => {
  it('should render successfully', () => {
    renderWithProviders(<ContactContainer />);
    expect(screen.getByText('Get In Touch')).toBeTruthy();
  });

  it('should render header section with correct content', () => {
    renderWithProviders(<ContactContainer />);

    expect(screen.getByText('Get In Touch')).toBeTruthy();
    expect(screen.getByText(/I'd love to hear from you/)).toBeTruthy();
  });

  it('should render contact form with all fields', () => {
    renderWithProviders(<ContactContainer />);

    expect(screen.getByText('Send Me a Message')).toBeTruthy();
    expect(screen.getByLabelText('Your Name')).toBeTruthy();
    expect(screen.getByLabelText('Your Email')).toBeTruthy();
    expect(screen.getByLabelText('Subject')).toBeTruthy();
    expect(screen.getByLabelText('Your Message')).toBeTruthy();
    expect(screen.getByRole('button', { name: /send message/i })).toBeTruthy();
  });

  it('should initialize form fields with empty values', () => {
    renderWithProviders(<ContactContainer />);

    expect(screen.getByLabelText('Your Name')).toHaveValue('');
    expect(screen.getByLabelText('Your Email')).toHaveValue('');
    expect(screen.getByLabelText('Subject')).toHaveValue('');
    expect(screen.getByLabelText('Your Message')).toHaveValue('');
  });

  it('should handle form input changes correctly', () => {
    renderWithProviders(<ContactContainer />);

    const nameInput = screen.getByLabelText('Your Name');
    const emailInput = screen.getByLabelText('Your Email');
    const subjectInput = screen.getByLabelText('Subject');
    const messageInput = screen.getByLabelText('Your Message');

    fireEvent.change(nameInput, { target: { value: 'John Doe' } });
    fireEvent.change(emailInput, { target: { value: 'john@example.com' } });
    fireEvent.change(subjectInput, { target: { value: 'Test Subject' } });
    fireEvent.change(messageInput, { target: { value: 'Test message' } });

    expect(nameInput).toHaveValue('John Doe');
    expect(emailInput).toHaveValue('john@example.com');
    expect(subjectInput).toHaveValue('Test Subject');
    expect(messageInput).toHaveValue('Test message');
  });

  it('should handle form submission and show success alert', async () => {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    const consoleSpy = jest.spyOn(console, 'log').mockImplementation(() => {});
    renderWithProviders(<ContactContainer />);

    // Fill out form
    fireEvent.change(screen.getByLabelText('Your Name'), {
      target: { value: 'John Doe' },
    });
    fireEvent.change(screen.getByLabelText('Your Email'), {
      target: { value: 'john@example.com' },
    });
    fireEvent.change(screen.getByLabelText('Subject'), {
      target: { value: 'Test Subject' },
    });
    fireEvent.change(screen.getByLabelText('Your Message'), {
      target: { value: 'Test message' },
    });

    // Submit form
    fireEvent.click(screen.getByRole('button', { name: /send message/i }));

    // Check that console.log was called with form data
    expect(consoleSpy).toHaveBeenCalledWith('Form submitted:', {
      name: 'John Doe',
      email: 'john@example.com',
      subject: 'Test Subject',
      message: 'Test message',
    });

    // Check that success alert is shown
    expect(
      screen.getByText("Thank you for your message! I'll get back to you soon.")
    ).toBeTruthy();

    // Check that form is reset
    expect(screen.getByLabelText('Your Name')).toHaveValue('');
    expect(screen.getByLabelText('Your Email')).toHaveValue('');
    expect(screen.getByLabelText('Subject')).toHaveValue('');
    expect(screen.getByLabelText('Your Message')).toHaveValue('');

    consoleSpy.mockRestore();
  });

  it('should render contact information section', () => {
    renderWithProviders(<ContactContainer />);

    expect(screen.getByText('Contact Information')).toBeTruthy();
    expect(screen.getByText('Email')).toBeTruthy();
    expect(screen.getByText('hello@portfolio.com')).toBeTruthy();
    expect(screen.getByText('Phone')).toBeTruthy();
    expect(screen.getByText('+1 (555) 123-4567')).toBeTruthy();
    expect(screen.getByText('Location')).toBeTruthy();
    expect(screen.getByText('San Francisco, CA')).toBeTruthy();
  });

  it('should render social links section', () => {
    renderWithProviders(<ContactContainer />);

    expect(screen.getByText('Follow Me')).toBeTruthy();

    // Check that social links have correct attributes
    const socialButtons = screen.getAllByRole('link');
    const linkedInButton = socialButtons.find(
      (button) =>
        button.getAttribute('href') === 'https://linkedin.com/in/portfolio'
    );
    const githubButton = socialButtons.find(
      (button) => button.getAttribute('href') === 'https://github.com/portfolio'
    );
    const twitterButton = socialButtons.find(
      (button) =>
        button.getAttribute('href') === 'https://twitter.com/portfolio'
    );

    expect(linkedInButton.getAttribute('target')).toBe('_blank');
    expect(githubButton.getAttribute('target')).toBe('_blank');
    expect(twitterButton.getAttribute('target')).toBe('_blank');
  });

  it('should render availability section', () => {
    renderWithProviders(<ContactContainer />);

    expect(screen.getByText('Availability')).toBeTruthy();
    expect(
      screen.getByText(/I'm currently available for freelance/)
    ).toBeTruthy();
  });

  it('should handle alert close correctly', async () => {
    renderWithProviders(<ContactContainer />);

    // Submit form to show alert
    fireEvent.click(screen.getByRole('button', { name: /send message/i }));

    // Wait for alert to appear
    await waitFor(() => {
      expect(
        screen.getByText(
          "Thank you for your message! I'll get back to you soon."
        )
      ).toBeTruthy();
    });

    // Close alert
    const closeButton = screen.getByRole('button', { name: /close/i });
    fireEvent.click(closeButton);

    // Alert should be closed
    await waitFor(() => {
      expect(
        screen.queryByText(
          "Thank you for your message! I'll get back to you soon."
        )
      ).not.toBeTruthy();
    });
  });

  it('should have correct CSS classes applied', () => {
    const { container } = renderWithProviders(<ContactContainer />);

    expect(container.querySelector('.contact')).toBeTruthy();
    expect(container.querySelector('.contact__header')).toBeTruthy();
    expect(container.querySelector('.contact__form')).toBeTruthy();
    expect(container.querySelector('.contact__sidebar')).toBeTruthy();
  });

  it('should not show success alert initially', () => {
    renderWithProviders(<ContactContainer />);
    expect(
      screen.queryByText(
        "Thank you for your message! I'll get back to you soon."
      )
    ).not.toBeTruthy();
  });

  it('should match snapshot', () => {
    const { container } = renderWithProviders(<ContactContainer />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
