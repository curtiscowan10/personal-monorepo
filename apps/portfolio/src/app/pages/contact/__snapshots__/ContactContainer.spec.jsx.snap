// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContactContainer > should match snapshot 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthLg contact css-5c1adp-MuiContainer-root"
>
  <div
    class="contact__header MuiBox-root css-0"
  >
    <h1
      class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom custom-title css-jp450z-MuiTypography-root"
    >
      Get In Touch
    </h1>
    <h5
      class="MuiTypography-root MuiTypography-h5 contact__subtitle css-rgpxvr-MuiTypography-root"
    >
      I'd love to hear from you! Whether you have a project in mind, want to collaborate, or just want to say hello, feel free to reach out.
    </h5>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-4 css-kqg16c-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root contact__form css-r6ekfc-MuiPaper-root-MuiCard-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiCardContent-root contact__form-content css-1lt5qva-MuiCardContent-root"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom contact__form-title css-18p8awu-MuiTypography-root"
          >
            Send Me a Message
          </h4>
          <form
            class="MuiBox-root css-0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-3 css-1u8xl8z-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-12qgy87-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r34:"
                    id=":r34:-label"
                  >
                    Your Name
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-47i793-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r34:"
                      name="name"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1n64csd-MuiNotchedOutlined-root"
                      >
                        <span>
                          Your Name
                           
                          *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-12qgy87-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r35:"
                    id=":r35:-label"
                  >
                    Your Email
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-47i793-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r35:"
                      name="email"
                      required=""
                      type="email"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1n64csd-MuiNotchedOutlined-root"
                      >
                        <span>
                          Your Email
                           
                          *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-12qgy87-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r36:"
                    id=":r36:-label"
                  >
                    Subject
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-47i793-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r36:"
                      name="subject"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1n64csd-MuiNotchedOutlined-root"
                      >
                        <span>
                          Subject
                           
                          *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-12qgy87-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r37:"
                    id=":r37:-label"
                  >
                    Your Message
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-ucdck6-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <textarea
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-w4nesw-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r37:"
                      name="message"
                      required=""
                      rows="6"
                      style="height: 0px; overflow: hidden;"
                    />
                    <textarea
                      aria-hidden="true"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-w4nesw-MuiInputBase-input-MuiOutlinedInput-input"
                      readonly=""
                      style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
                      tabindex="-1"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1n64csd-MuiNotchedOutlined-root"
                      >
                        <span>
                          Your Message
                           
                          *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary contact__form-submit css-dnximp-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="submit"
                >
                  Send Message
                  <span
                    class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeLarge css-1u6cks-MuiButton-endIcon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="SendIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
    >
      <div
        class="contact__sidebar MuiBox-root css-0"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root contact__info-card css-r6ekfc-MuiPaper-root-MuiCard-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <div
            class="MuiCardContent-root contact__info-card-content css-1lt5qva-MuiCardContent-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom contact__info-card-title css-wkkoyu-MuiTypography-root"
            >
              Contact Information
            </h5>
            <a
              class="contact__contact-item contact__contact-item--clickable MuiBox-root css-0"
              href="mailto:hello@portfolio.com"
            >
              <div
                class="contact__contact-icon MuiBox-root css-0"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-1umw9bq-MuiSvgIcon-root"
                  data-testid="EmailIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 4-8 5-8-5V6l8 5 8-5z"
                  />
                </svg>
              </div>
              <div
                class="MuiBox-root css-0"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-rzw81-MuiTypography-root"
                >
                  Email
                </h6>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1iw893a-MuiTypography-root"
                >
                  hello@portfolio.com
                </p>
              </div>
            </a>
            <a
              class="contact__contact-item contact__contact-item--clickable MuiBox-root css-0"
              href="tel:+15551234567"
            >
              <div
                class="contact__contact-icon MuiBox-root css-0"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-1umw9bq-MuiSvgIcon-root"
                  data-testid="PhoneIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02z"
                  />
                </svg>
              </div>
              <div
                class="MuiBox-root css-0"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-rzw81-MuiTypography-root"
                >
                  Phone
                </h6>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1iw893a-MuiTypography-root"
                >
                  +1 (555) 123-4567
                </p>
              </div>
            </a>
            <div
              class="contact__contact-item MuiBox-root css-0"
            >
              <div
                class="contact__contact-icon MuiBox-root css-0"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-1umw9bq-MuiSvgIcon-root"
                  data-testid="LocationOnIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5"
                  />
                </svg>
              </div>
              <div
                class="MuiBox-root css-0"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-rzw81-MuiTypography-root"
                >
                  Location
                </h6>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1iw893a-MuiTypography-root"
                >
                  San Francisco, CA
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root contact__social-card css-r6ekfc-MuiPaper-root-MuiCard-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <div
            class="MuiCardContent-root contact__social-card-content css-1lt5qva-MuiCardContent-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom contact__social-card-title css-wkkoyu-MuiTypography-root"
            >
              Follow Me
            </h5>
            <div
              class="contact__social-buttons MuiBox-root css-0"
            >
              <a
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium contact__social-button contact__social-button--linkedin css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
                href="https://linkedin.com/in/portfolio"
                rel="noopener noreferrer"
                tabindex="0"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="LinkedInIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                  />
                </svg>
              </a>
              <a
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium contact__social-button contact__social-button--github css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
                href="https://github.com/portfolio"
                rel="noopener noreferrer"
                tabindex="0"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="GitHubIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                  />
                </svg>
              </a>
              <a
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium contact__social-button contact__social-button--twitter css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
                href="https://twitter.com/portfolio"
                rel="noopener noreferrer"
                tabindex="0"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="TwitterIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root contact__availability-card css-r6ekfc-MuiPaper-root-MuiCard-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <div
            class="MuiCardContent-root contact__availability-card-content css-1lt5qva-MuiCardContent-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom contact__availability-card-title css-wkkoyu-MuiTypography-root"
            >
              Availability
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1k1cusp-MuiTypography-root"
            >
              I'm currently available for freelance projects and full-time opportunities. I typically respond to messages within 24 hours.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
